### PR TITLE
jinja2cpp: support conan v2, update dependencies

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **jinja2cpp/***

---

- [x] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [x] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/jinja2cpp/all/conanfile.py [hunk 1]
```diff
@@ -24,125 +26,101 @@ class Jinja2cppConan(ConanFile):
         "shared": False,
         "fPIC": True,
     }
-
-    generators = "cmake", "cmake_find_package"
-    _cmake = None
+    exports = ["CMakeLists.txt"]
```
Comment (memsharded) [inline recipes/jinja2cpp/all/conanfile.py:29]:
```suggestion
```
Not really exported anymore?

File: recipes/jinja2cpp/all/patches/fix-cmake-1.1.0.patch [hunk 2]
```diff
@@ -29,20 +31,37 @@
...
 +# - CONAN_PKG::rapidjson => rapidjson
 +# - CONAN_PKG::optional-lite => nonstd::optional-lite
 +# - CONAN_PKG::variant-lite => nonstd::variant-lite
-+set(JINJA2_PRIVATE_LIBS_INT ${CONAN_BOOST_PACKAGE_NAME} fmt::fmt CONAN_PKG::rapidjson)
-+set(JINJA2_PUBLIC_LIBS_INT nonstd::expected-lite CONAN_PKG::variant-lite CONAN_PKG::optional-lite nonstd::string-view-lite)
++set(JINJA2_PRIVATE_LIBS_INT Boost::headers Boost::filesystem fmt::fmt rapidjson)
++set(JINJA2_PUBLIC_LIBS_INT nonstd::expected-lite nonstd::variant-lite nonstd::optional-lite nonstd::string-view-lite)
+ 
+-set(JINJA2_PRIVATE_LIBS_INT ${CONAN_BOOST_PACKAGE_NAME} fmt::fmt rapidjson::rapidjson)
+-set(JINJA2_PUBLIC_LIBS_INT expected-lite::expected-lite variant-lite::variant-lite optional-lite::optional-lite string-view-lite::string-view-lite)
```
Comment (memsharded) [inline recipes/jinja2cpp/all/patches/fix-cmake-1.1.0.patch:67]: This changes could be probably submitted upstream?

File: recipes/jinja2cpp/all/conanfile.py [hunk 3]
```diff
@@ -24,125 +26,101 @@ class Jinja2cppConan(ConanFile):
...
+        tc.variables["JINJA2CPP_STRICT_WARNINGS"] = False
+        tc.variables["JINJA2CPP_BUILD_SHARED"] = self.options.shared
+        tc.variables["JINJA2CPP_DEPS_MODE"] = "conan-build"
+        tc.variables["JINJA2CPP_CXX_STANDARD"] = self.settings.compiler.get_safe("cppstd", 14)
         # Conan cmake generator omits the build_type flag for MSVC multiconfiguration CMake,
         # but provide build-type-specific runtime type flag. For now, Jinja2C++ build scripts
         # need to know the build type is being built in order to setup internal flags correctly
-        self._cmake.definitions["CMAKE_BUILD_TYPE"] = self.settings.build_type
-        if self._is_msvc:
+        tc.variables["CMAKE_BUILD_TYPE"] = self.settings.build_type
```
Comment (czoido) [inline recipes/jinja2cpp/all/conanfile.py:94]: I think we can remove this

**Review by czoido — APPROVED** [review]
Made some minor changes:
- Removing export of CMakeLists.txt not used
- Remove setting CMAKE_BUILD_TYPE in recipe

**Review by SSE4 — APPROVED** [review]

**Review by memsharded — APPROVED** [review]

**Pull request merged**